### PR TITLE
Hw 161

### PR DESCRIPTION
**Pull request by TarasHryts**

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
   </component>
+  <component name="MavenProjectsManager">
+    <option name="originalFiles">
+      <list>
+        <option value="$PROJECT_DIR$/pom.xml" />
+      </list>
+    </option>
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="false" project-jdk-name="11" project-jdk-type="JavaSDK" />
 </project>
```
Comment (Sofasmile) [inline .idea/misc.xml]: add a blank line at the end of the file

File: pom.xml [hunk 2]
```diff
@@ -26,13 +27,30 @@
...
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>jstl</artifactId>
+            <version>1.2</version>
+        </dependency>
+
```
Comment (Sofasmile) [inline pom.xml]: remove redundant lines

File: pom.xml [hunk 3]
```diff
@@ -26,13 +27,30 @@
...
+
+
     </dependencies>
 
     <build>
-        <plugins>
+        <!--<plugins>
```
Comment (Sofasmile) [inline pom.xml]: your repository shouldn't consist commented code

File: src/main/java/mate/academy/shop/Main.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+//package mate.academy.shop;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/Main.java]: better remove this file
 or remove from repository

File: src/main/java/mate/academy/shop/anotation/Injector.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                            || implementation.getClass().getAnnotation(Service.class) != null) {
+                        field.setAccessible(true);
+                        field.set(null, implementation);
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * Scans all classes accessible from the context class loader which belong to the given package and subpackages.
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/anotation/Injector.java]: too long line

File: src/main/java/mate/academy/shop/anotation/Injector.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    /**
+     * Recursive method used to find all classes in a given directory and subdirs.
+     *
+     * @param directory   The base directory
+     * @param packageName The package name for classes found inside the base directory
+     * @return The classes
+     * @throws ClassNotFoundException
+     */
+    private static List<Class> findClasses(File directory, String packageName) throws ClassNotFoundException {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/anotation/Injector.java]: https://mate-academy.github.io/style-guides/java/java.html#s4.4-column-limit

File: src/main/java/mate/academy/shop/anotation/Injector.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            return classes;
+        }
+        File[] files = directory.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    assert !file.getName().contains(".");
+                    classes.addAll(findClasses(file, packageName + "." + file.getName()));
+                } else if (file.getName().endsWith(".class")) {
+                    classes.add(Class.forName(packageName + '.' + file.getName().substring(0, file.getName().length() - 6)));
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/anotation/Injector.java]: :)

File: src/main/java/mate/academy/shop/controller/AddToBucketController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class AddToBucketController extends HttpServlet {
+    @Inject
+    private static BucketService bucketService;
+    @Override
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/controller/AddToBucketController.java]: better add a blank line between method and feeld

File: src/main/java/mate/academy/shop/controller/AddToBucketController.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Bucket bucket = bucketService.get(0L);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/controller/AddToBucketController.java]: is the 0L the magic number?
in all controllers)

File: src/main/java/mate/academy/shop/controller/AddToBucketController.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+        String itemId = req.getParameter("item_id");
+        bucketService.addItem(0L, Long.valueOf(itemId));
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/controller/AddToBucketController.java]: let's use only userId as a known parameter, without other constants in all controllers

File: src/main/java/mate/academy/shop/controller/InjectInitializer.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        items.add(new Item("Moskvich", 1D));
+        items.add(new Item("Opel", 58D));
+        items.add(new Item("BMW", 2210D));
+        items.add(new Item("Reno", 50D));
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {
+    }
+}
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/controller/InjectInitializer.java]: add a blank line

File: src/main/java/mate/academy/shop/model/Item.java [hunk 12]
```diff
@@ -14,19 +14,19 @@ public Item(String name, Double price) {
     }
 
     public Long getId() {
-        return id;
+        return this.id;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/model/Item.java]: do we need `this` in getters?

File: src/main/java/mate/academy/shop/service/UserService.java [hunk 13]
```diff
@@ -1,14 +1,19 @@
...
     User create(User user);
+
     User get(Long id);
+
     List<Order> getAllOrders(Long userId);
+
     User update(User user);
+
     User delete(Long id);
+    List<User> getAll();
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/service/UserService.java]: add a blank line between all methods

File: src/main/java/mate/academy/shop/service/impl/BucketServiceImpl.java [hunk 14]
```diff
@@ -27,6 +28,11 @@ public Bucket get(Long id) {
         return bucketDao.get(id);
     }
 
+    @Override
+    public Bucket getLast() {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/service/impl/BucketServiceImpl.java]: do we need this method?

File: src/main/java/mate/academy/shop/service/impl/ItemServiceImpl.java [hunk 15]
```diff
@@ -35,4 +38,9 @@ public Item delete(Long id) {
     public Item delete(Item item) {
         return itemDao.delete(item);
     }
+
+    @Override
+    public List<Item> getAll() {
+        return Storage.items;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/service/impl/ItemServiceImpl.java]: we cant use db in service layer

File: src/main/java/mate/academy/shop/service/impl/UserServiceImpl.java [hunk 16]
```diff
@@ -38,4 +39,9 @@ public User update(User user) {
     public User delete(Long id) {
         return userDao.delete(id);
     }
+
+    @Override
+    public List<User> getAll() {
+        return Storage.users;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/service/impl/UserServiceImpl.java]: the same

File: .idea/misc.xml (hunk 1 above)
Comment (TarasHryts) [reply .idea/misc.xml]: done

File: pom.xml (hunk 2 above)
Comment (TarasHryts) [reply pom.xml]: done

File: src/main/java/mate/academy/shop/Main.java (hunk 4 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/Main.java]: deleted

File: src/main/java/mate/academy/shop/anotation/Injector.java (hunk 5 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/anotation/Injector.java]: done

File: src/main/java/mate/academy/shop/model/Item.java (hunk 12 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/model/Item.java]: done

File: src/main/java/mate/academy/shop/service/UserService.java (hunk 13 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/service/UserService.java]: done

File: src/main/java/mate/academy/shop/service/impl/BucketServiceImpl.java (hunk 14 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/service/impl/BucketServiceImpl.java]: I needed this method to retrieve the last created user basket but gave up this implementation. I will delete it

File: src/main/java/mate/academy/shop/service/impl/ItemServiceImpl.java (hunk 15 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/service/impl/ItemServiceImpl.java]: solved

File: src/main/java/mate/academy/shop/service/impl/UserServiceImpl.java (hunk 16 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/service/impl/UserServiceImpl.java]: solved. For faster work, i turned off CheckStyle.
Sorry.

File: .idea/misc.xml [hunk 17]
```diff
@@ -1,6 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (Sofasmile) [inline .idea/misc.xml]: oh sorry
remove .idea/* from your repository

File: src/main/java/mate/academy/shop/anotation/Injector.java (hunk 5 above)
Comment (Sofasmile) [reply src/main/java/mate/academy/shop/anotation/Injector.java]: ?

File: src/main/java/mate/academy/shop/controller/AddToBucketController.java (hunk 9 above)
Comment (Sofasmile) [reply src/main/java/mate/academy/shop/controller/AddToBucketController.java]: потрібно винести всі '0L' в константи у всіх контроллерах

File: src/main/java/mate/academy/shop/controller/InjectInitializer.java (hunk 11 above)
Comment (Sofasmile) [reply src/main/java/mate/academy/shop/controller/InjectInitializer.java]: ?

File: .idea/misc.xml (hunk 17 above)
Comment (TarasHryts) [reply .idea/misc.xml]: I tried, but almost broke a project

Comment (TarasHryts) [reply .idea/misc.xml]: @Override
    public void contextDestroyed(ServletContextEvent servletContextEvent) {
    }
it is similar to before and after method in tests, and it shoud be implement

File: pom.xml [hunk 18]
```diff
@@ -42,6 +42,28 @@
     </dependencies>
 
     <build>
+        <plugins>
+<!--            <plugin>-->
```
Comment (Sofasmile) [inline pom.xml]: remove this commented code

**Pull request merged**